README.md

```diff
@@ -519,17 +519,26 @@ Where:
 * `REASON` is an arbitrary error reason term (in case of `status="error"`) or an empty string (when `status="success"`)
 * `LE` defines the `upper inclusive bound` (`less than or equal`) values for buckets, currently `1000`, `10_000`, `25_000`, `50_000`, `100_000`, `250_000`, `500_000`, `1000_000` or `+Inf`
 
+This histogram metric shows the distribution of times needed to:
+1. Select a worker (this may include waiting time when all workers are busy).
+2. Send a request.
+3. Get a response from push notifications provider.
+
+###### HTTP/2 requests
```

Comment:
Does this metric show time spent only on sending the request and waiting for the response?

Reply:
This metric shows only the time it takes to handle and send request. It only opens the stream and sends the request. It does not measure response time. In general, when this timer is small it it possible that we are getting errors with opening the stream.

Reply:
Thanks for checking this. I think it'd be good to add this info to the doc.

Reply:
Ok, will add that info.

Reply:
@michalwski, this part is already added to the doc..

Reply:
Hmm, I wonder how did you come to the conclusion that it measures only the time it takes to handle and send the request. If I'm correct this metric measures the time of this function in sparrow: https://github.com/esl/sparrow/blob/master/lib/sparrow/h2_worker.ex#L352-L430

Inside the function, we are waiting for the response and emit additional events depending on the response.

Also, I think the name of the metric starts with `sparrow_h2_worker` instead of `sparrow_h_worker`.

Reply:
All right, we discussed this offline with @janciesla8818. I was misled by the `post_result` var name and the case checking if this is a successful response or not. In fact, this var gives us information if the request was sent successfully or not. The response to the request arrives asynchronously (HTTP/2). Thanks to that we can send many requests using the same `h2_worker` and connection.

When it comes to the metric name, the `Prometheus.Core` lib removes all numbers from the metric name.